rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -199,15 +205,23 @@ void RecorderImpl::stop()
   {
     std::lock_guard<std::mutex> lock(event_publisher_thread_mutex_);
     event_publisher_thread_should_exit_ = true;
+    event_publisher_thread_wake_cv_.notify_all();
```

Comment:
I don't think this change is necessary. According to [the docs](https://en.cppreference.com/w/cpp/thread/condition_variable/notify_all):
> The notifying thread does not need to hold the lock on the same mutex as the one held by the waiting thread(s). Doing so may be a pessimization,

Reply:
You are right. It seems I missed a fact that we are checking for `event_publisher_thread_should_exit_` in the `condition_variable::wait(..)` under the mutex. 
And if it is true even before we will call for the `wait(..)` for the first time. The `wait()..` will exit immediately even without `notify_all()` call.
I will move `event_publisher_thread_wake_cv_.notify_all();` out of the mutex lock.